cmake/ConfigureCompiler.cmake

```diff
@@ -1,25 +1,22 @@
-function(env_set var_name default_value type docstring)
-  set(val ${default_value})
-  if(DEFINED ENV{${var_name}})
-    set(val $ENV{${var_name}})
-  endif()
-  set(${var_name} ${val} CACHE ${type} "${docstring}")
-endfunction()
+include(CompilerChecks)
 
-set(USE_GPERFTOOLS OFF CACHE BOOL "Use gperfools for profiling")
+env_set(USE_GPERFTOOLS OFF BOOL "Use gperfools for profiling")
 env_set(USE_VALGRIND OFF BOOL "Compile for valgrind usage")
-set(USE_VALGRIND_FOR_CTEST ${USE_VALGRIND} CACHE BOOL "Use valgrind for ctest")
-set(ALLOC_INSTRUMENTATION OFF CACHE BOOL "Instrument alloc")
-set(WITH_UNDODB OFF CACHE BOOL "Use rr or undodb")
-set(USE_ASAN OFF CACHE BOOL "Compile with address sanitizer")
-set(USE_UBSAN OFF CACHE BOOL "Compile with undefined behavior sanitizer")
-set(FDB_RELEASE OFF CACHE BOOL "This is a building of a final release")
-env_set(USE_LD "DEFAULT" STRING "The linker to use for building: can be LD (system default, default choice), BFD, GOLD, or LLD")
-env_set(USE_LIBCXX OFF BOOL "Use libc++")
+env_set(USE_VALGRIND_FOR_CTEST ${USE_VALGRIND} BOOL "Use valgrind for ctest")
+env_set(ALLOC_INSTRUMENTATION OFF BOOL "Instrument alloc")
+env_set(WITH_UNDODB OFF BOOL "Use rr or undodb")
+env_set(USE_ASAN OFF BOOL "Compile with address sanitizer")
+env_set(USE_UBSAN OFF BOOL "Compile with undefined behavior sanitizer")
+env_set(FDB_RELEASE OFF BOOL "This is a building of a final release")
 env_set(USE_CCACHE OFF BOOL "Use ccache for compilation if available")
-set(RELATIVE_DEBUG_PATHS OFF CACHE BOOL "Use relative file paths in debug info")
-set(STATIC_LINK_LIBCXX ON CACHE BOOL "Statically link libstdcpp/libc++")
-set(USE_WERROR OFF CACHE BOOL "Compile with -Werror. Recommended for local development and CI.")
+env_set(RELATIVE_DEBUG_PATHS OFF BOOL "Use relative file paths in debug info")
+env_set(USE_WERROR OFF BOOL "Compile with -Werror. Recommended for local development and CI.")
+default_linker(_use_ld)
+env_set(USE_LD "${_use_ld}" STRING "The linker to use for building: can be LD (system default, default choice), BFD, GOLD, or LLD")
```

Comment:
IIUC LD is no longer the default choice if LLD is found, please update the description